games/xworld/maps/xworld_env.py

```diff
@@ -80,12 +84,23 @@ def __init__(self, item_path, max_height=7, max_width=7):
                             for l in lines if not l.startswith("//") and not l == ""}
 
     ############################ interface with Python tasks ############################
-    def reset(self):
+    def reset(self, same_session=False):
         """
```

Comment:
Move this logic to your own map file; use derived class members to record the state of map changes.

Reply:
Done.